Magento2/Sniffs/Annotation/MethodArgumentsSniff.php

```diff
@@ -617,6 +622,104 @@ public function process(File $phpcsFile, $stackPtr)
         );
     }
 
+    /**
+     * Check method if it has defined return & parameter types, then it doesn't need DocBlock
+     *
+     * @param File $phpcsFile
+     * @param int $stackPtr
+     * @return bool
+     */
+    private function checkIfMethodNeedsDocBlock(File $phpcsFile, $stackPtr) : bool
+    {
+        if ($this->checkIfNamespaceContainsApi($phpcsFile)) {
+            // Note: API classes MUST have DocBlock because it uses them instead of reflection for types
+            return true;
+        }
+        $foundAllParameterTypes = true;
+        $methodParameters = $phpcsFile->getMethodParameters($stackPtr);
+        foreach ($methodParameters as $parameter) {
+            if (!$parameter['type_hint']) {
+                return true; // doesn't have type hint
+            }
+        }
+        $methodProperties = $phpcsFile->getMethodProperties($stackPtr);
+        $foundReturnType = !!$methodProperties['return_type'];
+        if (!$foundReturnType) {
+            $methodName = $phpcsFile->getDeclarationName($stackPtr);
+            // Note: __construct and __destruct can't have return types
+            if ('__construct' !== $methodName && '__destruct' !== $methodName) {
+                return true;
+            }
+        }
+        $complexity = $this->getMethodComplexity($phpcsFile, $stackPtr);
+        return $complexity > static::MAXIMUM_COMPLEXITY_ALLOWED_FOR_NO_DOCBLOCK;
+    }
+
+    /**
+     * Check if namespace contains API
+     *
+     * @param File $phpcsFile
+     * @return bool
+     */
+    private function checkIfNamespaceContainsApi(File $phpcsFile) : bool
+    {
+        $namespaceStackPtr = $phpcsFile->findNext(T_NAMESPACE, 0);
+        if (!is_int($namespaceStackPtr)) {
+            return false;
+        }
+        $tokens = $phpcsFile->getTokens();
+        // phpcs:ignore Squiz.ControlStructures.ForLoopDeclaration.SpacingAfterOpen,Generic.CodeAnalysis.ForLoopWithTestFunctionCall.NotAllowed
```

Comment:
Why not respect the coding standard?
```suggestion
```

Reply:
For Generic.CodeAnalysis.ForLoopWithTestFunctionCall.NotAllowed: 
As you may know, since PHP 7.4, array_key_exists is no longer a PHP function.  It is compile to its own opcode, which has dramatically increased its performance.  Calling array_key_exists is O(1) since PHP arrays are implemented as hash tables which have O(1) CPU time complexity for lookups.
https://www.php.net/manual/en/migration74.other-changes.php
```A specialized VM opcode for the [array_key_exists()](https://www.php.net/manual/en/function.array-key-exists.php) function has been added, which improves performance of this function if it can be statically resolved. In namespaced code, this may require writing \array_key_exists() or explicitly importing the function```
Though, it looks like I messed up because I forgot to either use the `\` prefix or import the function.

For Squiz.ControlStructures.ForLoopDeclaration.SpacingAfterOpen:
Because the line that starts the for loop became over 120 characters long, I had to separate it into multiple lines.

Reply:
I've resolved these in 17e120ff553a9b5123c3b11602cdb98491293fe6 by replacing array_key_exists with the null coalescing operator.